File: wup1/Java/Front3.java
/* 
    front3

    Given a string, we'll say that the front is the first 3 chars of the string. 
    If the string length is less than 3, the front is whatever is there. 
    Return a new string which is 3 copies of the front.


*/

public class Front3 {
    
    public static void main(String[] args) {
        Front3 f3 = new Front3();
        f3.test();
	}

    public String front3(String str) {
        if (str.length() > 3)
            return str.substring(0,3) + str.substring(0,3) + 
                str.substring(0,3);
        else
            return str + str + str;
    }

    public void test() {
        assert front3("Java").equals("JavJavJav");
        assert front3("Chocolate").equals("ChoChoCho");
        assert front3("abc").equals("abcabcabc");
        assert front3("abcXYZ").equals("abcabcabc");
        assert front3("ab").equals("ababab");
        assert front3("a").equals("aaa");
        assert front3("").equals("");
        System.out.println("All tests passed");
    }
}

File: wup1/JS/front3.js
/*
Warmup-1 --

front3

Given a string, we'll say that the front is the first 3 chars of the string. 
If the string length is less than 3, the front is whatever is there. 
Return a new string which is 3 copies of the front.

Examples

    front3('Java') → JavJavJav
    front3('Chocolate') → ChoChoCho
    front3('abc') → abcabcabc
*/

function front3(str){
  if (str.length === 3) {
    return str + str + str;
  }
  let first = str.substring(0,3);
  return first + first + first;
}

const assert = require('assert');
assert.equal(front3('Java'), 'JavJavJav');
assert.equal(front3('Chocolate'), 'ChoChoCho');
assert.equal(front3('abc'), 'abcabcabc');
assert.equal(front3('abcXYZ'), 'abcabcabc');
assert.equal(front3('ab'), 'ababab');
assert.equal(front3('a'), 'aaa');
assert.equal(front3(''), '');
console.log("All tests passed");
